### ISSUE TITLE
Gracefully handle certain VirtualBox errors when VMs fail to come up

### ISSUE THREAD

**Issue by ianvernon** [cilium/cilium#2172]
VirtualBox VMs fail to come up when many VMs are launched simultaneously on the same Jenkins agent:
```
16:34:31 [cilium_cilium_master-7EP22SMIGRU3LKKBKJ6LHNBOPNZYHH476YWIUKJLRL2MTFOGJDFQ@2] Running shell script
16:34:31 + CILIUM_USE_ENVOY=1 ./contrib/vagrant/start.sh
16:34:33 Bringing machine 'cilium-master-build-master-909-envoy' up with 'virtualbox' provider...
16:34:33 ==> cilium-master-build-master-909-envoy: Importing base box 'cilium/ubuntu-16.10'...
16:35:30 
[KProgress: 10%
[KProgress: 20%
[KProgress: 30%
[KProgress: 40%
[KProgress: 50%
[KProgress: 60%
[KProgress: 70%
[KProgress: 90%
[K==> cilium-master-build-master-909-envoy: Matching MAC address for NAT networking...
16:35:31 ==> cilium-master-build-master-909-envoy: Checking if box 'cilium/ubuntu-16.10' is up to date...
16:35:32 ==> cilium-master-build-master-909-envoy: Setting the name of the VM: cilium_cilium_master-7EP22SMIGRU3LKKBKJ6LHNBOPNZYHH476YWIUKJLRL2MTFOGJDFQ2_cilium-master-build-master-909-envoy_1511973331730_77165
16:35:32 ==> cilium-master-build-master-909-envoy: Fixed port collision for 22 => 2222. Now on port 2201.
16:35:32 ==> cilium-master-build-master-909-envoy: Clearing any previously set network interfaces...
16:35:32 ==> cilium-master-build-master-909-envoy: Preparing network interfaces based on configuration...
16:35:32     cilium-master-build-master-909-envoy: Adapter 1: nat
16:35:32     cilium-master-build-master-909-envoy: Adapter 2: intnet
16:35:32 There was an error while executing `VBoxManage`, a CLI used by Vagrant
16:35:32 for controlling VirtualBox. The command and stderr is shown below.
16:35:32 
16:35:32 Command: ["modifyvm", "80b151c5-187b-4051-bc43-fc3bae5b0413", "--nic1", "nat", "--nic2", "intnet", "--intnet2", "cilium-test-master-909-envoy", "--cableconnected2", "on"]
16:35:32 
16:35:32 Stderr: VBoxManage: error: The machine 'cilium_cilium_master-7EP22SMIGRU3LKKBKJ6LHNBOPNZYHH476YWIUKJLRL2MTFOGJDFQ2_cilium-master-build-master-909-envoy_1511973331730_77165' is already locked for a session (or being unlocked)
16:35:32 VBoxManage: error: Details: code VBOX_E_INVALID_OBJECT_STATE (0x80bb0007), component MachineWrap, interface IMachine, callee nsISupports
16:35:32 VBoxManage: error: Context: "LockMachine(a->session, LockType_Write)" at line 507 of file VBoxManageModifyVM.cpp
```

Figure out a way of gracefully handling these errors and recovering from them if possible so that the build doesn't fail. Alternatively, we can try to use information available from `VBoxManage` or `vagrant` commands to try and see how long VMs have been 'running' before, and if a lot of VMs are in 'running' state and have been started recently, back off and try again until those VMs have been provisioned or have been booted for some time.

**Comment by joestringer** [comment]
Can we work around this by using something like `VBoxManage list intnets | grep intnet | wc -l` or `VBoxManage list natnets | grep nat | wc -l` to wait for the relevant networks to be configured?

**Comment by joestringer** [comment]
Similar issue again on master: https://jenkins.cilium.io/job/cilium/job/cilium/job/master/920/execution/node/34/log/

**Comment by ianvernon** [comment]
Fixed by #2228, in which VMs are booted in serial order, and then provisioned in parallel per build.
